batteryInterfaceSource.cpp

```diff
@@ -0,0 +1,161 @@
+#pragma once
+#include <assert.h>
+#include <iostream>
+#include "batteryInterface.h"
+#include <iomanip>
+#include <sstream>  
+#include <map>
+#include <utility>
+using namespace std;
+
+int BatteryElements::temperature;
+float BatteryElements::voltage;
+float BatteryElements::current;
+
+double CurrentIndicator::currentMinThreshould = 0.8;
+double CurrentIndicator::currentMaxThreshould = 1.2;
+
+float VoltageIndicator::voltageMinThreshould = 3;
+float VoltageIndicator::vOltageMaxThreshould = 6;
+
+float TempratureIndicator::temperatureMinThreshould = -20;
+float TempratureIndicator::temperatureMaxThreshould = 50;
+
+int   StateOfChargeRate::fullBatteryStatus = 80;
+int   StateOfChargeRate::lowBatteryStatus = 20;
+
+int   StatusOfCharge::currentBatteryStatus;
+float weatherIndicator::todaysTemperature;
+float weatherIndicator::hotWeather  = 70;
+float weatherIndicator::coldWeather = -50;
+
+string LanguageSupported::language;
+LanguageSupported languagesupported;
+
+
+void BatterySpecification::BatterySpecificationPrinter()
+{
+	cout << "    || "<< languagesupported.status[{"W_status", LanguageSupported::language}]<<"|| " << "           ||"<< languagesupported.status[{"B_status", LanguageSupported::language}]<<"|| " << "                         ||"<< languagesupported.status[{"B_current_voltage", LanguageSupported::language}] <<"||" << endl;

```

Comment:
Nice separation of printing from processing :)
Just that it's hard for another person to add a parameter to print and be confident that it will work. Maybe it will require seeing the output and doing some trial-and-error. Perhaps you did too?
Any way to make this easier?

Reply:
yes, i try to make it easier.